File: backend/app/Services/NotificationService.php
<?php

namespace App\Services;

use App\Models\Notification;
use App\Repositories\NotificationRepository;

class NotificationService
{
    public function __construct(private NotificationRepository $repo) {}

    public function create(array $data): Notification
    {
        $data['user_id'] = auth()->id();
        return $this->repo->create($data);
    }

    public function getUnread()
    {
        return $this->repo->getUnread(auth()->id());
    }

    public function getAll(int $perPage = 20)
    {
        return $this->repo->paginate(auth()->id(), $perPage);
    }

    public function markRead(int $id): Notification
    {
        return $this->repo->markRead($id, auth()->id());
    }

    public function markAllRead(): int
    {
        return $this->repo->markAllRead(auth()->id());
    }

    public function unreadCount(): int
    {
        return $this->repo->unreadCount(auth()->id());
    }
}

File: backend-dotnet/Services/NotificationService.cs
using TexTrack.Api.Auth;
using TexTrack.Api.Infrastructure;
using TexTrack.Api.Models;
using TexTrack.Api.Repositories;

namespace TexTrack.Api.Services;

public class NotificationService
{
    private readonly NotificationRepository _repo;
    private readonly CurrentUser _currentUser;

    public NotificationService(NotificationRepository repo, CurrentUser currentUser)
    {
        _repo = repo;
        _currentUser = currentUser;
    }

    public Task<Notification> CreateAsync(string title, string message, string type)
    {
        var notification = new Notification
        {
            UserId = _currentUser.Id,
            Title = title,
            Message = message,
            Type = type,
        };
        return _repo.CreateAsync(notification);
    }

    public Task<PagedResult<Notification>> GetAllAsync(int perPage = 20) =>
        _repo.PaginateAsync(_currentUser.Id ?? 0, perPage);

    public Task<Notification> MarkReadAsync(long id) =>
        _repo.MarkReadAsync(id, _currentUser.Id ?? 0);

    public Task<int> MarkAllReadAsync() =>
        _repo.MarkAllReadAsync(_currentUser.Id ?? 0);

    public Task<int> UnreadCountAsync() =>
        _repo.UnreadCountAsync(_currentUser.Id ?? 0);
}
